pandas/core/construction.py

```diff
@@ -548,6 +549,10 @@ def sanitize_array(
         # Avoid ending up with a NumpyExtensionArray
         dtype = dtype.numpy_dtype
 
+    object_index = False
+    if isinstance(data, ABCIndex) and data.dtype == object and dtype is None:
```

Comment:
We don't have the same issue with `Series[object]`?

Reply:
No Series takes an early exit, brock and I chatted about deprecating this behavior for Index altogether as well